File: e-learning-service/els-graphql/src/main/java/org/example/elsgraphql/service/UserService.java
package org.example.elsgraphql.service;

import org.example.elsgraphql.model.User;

import java.util.Optional;

public interface UserService {
    User createUser(String name, String email, String password);

    Optional<User> findById(Long id);

    User updateUser(Long id, String name, String email);
}

File: kotlin/kotlin-vs-java/java-example/src/main/kotlin/org/example/user/UserService.kt
package org.example.user

import org.example.model.UserDto
import java.time.LocalDateTime

class UserService {

    fun logic(userDto: UserDto? = null) {
        val userDto = UserDto(
            null,
            20,
            "email@email.com",
            "01012341234",
            LocalDateTime.now(),
        )

        userDto.name?.let {
            println(it)
        } ?: println("null")

        println(userDto)
    }
}

fun main() {
    UserService().logic()
}